shared-module/bitbangio/SPI.c

```diff
@@ -123,7 +123,7 @@ void shared_module_bitbangio_spi_unlock(bitbangio_spi_obj_t *self) {
 // Writes out the given data.
 bool shared_module_bitbangio_spi_write(bitbangio_spi_obj_t *self, const uint8_t *data, size_t len) {
     if (len > 0 && !self->has_mosi) {
-        mp_raise_ValueError(translate("Cannot write without MOSI pin."));
+        mp_raise_ValueError(translate("Cannot write without MOSI pin"));
```

Comment:
can make this match a string elsewhere

Reply:
Interestingly, some ports test for writes without MOSI or reads without MISO, and some do not. For instance, on `atmel-samd`, a write without MOSI does not raise an error.